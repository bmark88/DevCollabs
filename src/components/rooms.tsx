import React, { ReactNode } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import styled from "styled-components";

interface Props {
  children: ReactNode
  image :string
  title :string
}

const useStyles = makeStyles({
  root: {
    backgroundColor: '#EEE',
    maxWidth: 240,
    minWidth: 240,
    marginTop: 0,
    margin: 15,
    ['@media (max-width:1200px)']: {
      maxWidth: 240,
      minWidth: 240,
    },
    ['@media (max-width: 620px)']: {
      maxWidth: 260,
      minWidth: 260,
    }
  },
  font: {
    fontSize: '18px',
  },
  media: {
    height: 140,
  },
});

const RoomContainer = styled.div`
  margin: 1em;
  display: flex;
  justify-content: center;
  min-width: 330px;

  @media (max-width: 620px) {
    margin: 0;
    padding: 0;
  }
  
  @media (max-width: 1880px) {
    flex-wrap: wrap;
    justify-content: center;
  }
`;

const Div = styled.div`
  display: flex;
  justify-content: center;
  padding: 0.5em;
  margin-bottom: -0.25em;
  font-size: 20px;
`;

const RoomCard = ({ children, image, title } :Props) => {
  const classes = useStyles();

  return (
    <Card className={`${classes.root} ${"dark"}`}>  
        <CardMedia 
          className={`${classes.media} ${"dark"}`}
          image={image}
        />
        <CardContent className="dark">
          <Typography gutterBottom align="center" variant="h5" component="h2" className={classes.font}>
            {title}
          </Typography>
        </CardContent>
        <Div className={`${classes.font} ${"dark"}`}>
          {children}
        </Div>
    </Card>
  );
}

export { RoomCard, RoomContainer }