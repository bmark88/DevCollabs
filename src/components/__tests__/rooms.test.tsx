import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"

import { RoomCard } from "../rooms"

describe("LoginForm", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
      <RoomCard               
        image="https://sociorocketnewsen.files.wordpress.com/2014/01/anonymous.jpg?w=580&h=350"
        title="Create A Group">
      </RoomCard>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  });

  it("should contain a room card title", () => {
    const { getByText } = render(
      <RoomCard               
        image="https://sociorocketnewsen.files.wordpress.com/2014/01/anonymous.jpg?w=580&h=350"
        title="Create A Group">
      </RoomCard>)

      expect(getByText("Create A Group")).toBeDefined();
  });
});