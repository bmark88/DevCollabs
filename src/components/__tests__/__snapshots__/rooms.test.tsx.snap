// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root makeStyles-media-3"
    style={
      Object {
        "backgroundImage": "url(\\"https://sociorocketnewsen.files.wordpress.com/2014/01/anonymous.jpg?w=580&h=350\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root makeStyles-font-2 MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Create A Group
    </h2>
  </div>
  <div
    className="sc-AxirZ gRXEyv makeStyles-font-2"
  />
</div>
`;
